modules/engines/engine-javascript-graalium/src/main/java/org/eclipse/dirigible/engine/js/graalium/execution/platform/GraalJSFileSystem.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2022 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.dirigible.engine.js.graalium.execution.platform;
+
+import org.eclipse.dirigible.commons.api.scripting.ScriptingException;
+import org.eclipse.dirigible.engine.js.graalium.execution.dependencies.DependencyProvider;
+import org.eclipse.dirigible.engine.js.graalium.execution.modules.DirigibleCoreModuleResolver;
+import org.graalvm.polyglot.io.FileSystem;
+
+import java.io.*;
+import java.net.URI;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.file.*;
+import java.nio.file.attribute.FileAttribute;
+import java.nio.file.spi.FileSystemProvider;
+import java.util.*;
+
+public class GraalJSFileSystem implements FileSystem {
+
+    private static final FileSystemProvider DELEGATE = FileSystems.getDefault().provider();
+    private Path currentWorkingDirectoryPath;
+    private final DirigibleCoreModuleResolver dirigibleCoreModuleResolver;
+    private final DependencyProvider dependencyProvider;
+
+    public GraalJSFileSystem(Path currentWorkingDirectoryPath, DirigibleCoreModuleResolver dirigibleCoreModuleResolver) {
+        this.currentWorkingDirectoryPath = currentWorkingDirectoryPath;
+        this.dirigibleCoreModuleResolver = dirigibleCoreModuleResolver;
+        this.dependencyProvider = new DependencyProvider();
+    }
+
+    @Override
+    public Path parsePath(URI uri) {
+        return dependencyProvider.provideDependency(uri);
+    }
+
+    /**
+     * Given a path string returns a absolute path relative to the CWD
+     */
+    @Override
+    public Path parsePath(String path) {
+        if ("".equals(path)) {
+            return currentWorkingDirectoryPath;
+        }
+
+        if (dirigibleCoreModuleResolver.isCoreModule(path)) {
+            return dirigibleCoreModuleResolver.resolveCoreModulePath(path);
+        }
+
+        return Path.of(path);
```

Comment:
It's a good practice to have only a single `return` statement per method _(of course wherever is possible only)_.